src/airbrakes.py

```diff
@@ -1,6 +1,181 @@
+
+
+"""
+This class implements a Kalman filter to determine how far the airbrakes
+flaps should be deployed, based on sensor data and previous flight data.
+
+
+Attributes
+----------
+
+
+
+Methods
+-------
+
+
+
+
+"""
 class KalmanFilter:
   def __init__(self, rd):
     rocket_data = rd
+
+
+
+"""
+
+Airbrakes: This class provides a simple way to interface with the airbrakes
+hardware. It is used by the KalmanFilter class.
+
+
+Attributes
+----------
+
+__max_pot_val: Value of the potentiometer as read by the ADC when brakes are
+fully deployed. Set by calibrate().
+
+__min_pot_val: Value of the potentiometer as read by the ADC when brakes are
+fully closed. Set by calibrate().
+
+pot_pin: ADS1115 pin that is connected to potentiometer
+step_pin: GPIO pin that controls stepping of A4988 stepper driver
+dir_pin: GPIO pin that controls direction of A4988 stepper driver
+
+motor_direction: If motor is turning the wrong direction when testing,
+invert the value of this (True/False).
+
+step_delay: Delay in seconds between pulses of step_pin
+step_angle: Step angle in degrees of stepper motor
+microsteps: 2 (half), 4 (quarter), 16 (sixteenth) etc.
+gear_ratio: (# of motor turns) / (# of gearbox output turns)
+
+__max_steps_to_open: Maximum numper of steps needed to fully open the brakes.
+This is calculated from the motor parameters: step_angle, microsteps, gear_ratio
+
+
+Methods
+-------
+
+__singleStep():
+  Steps Airbrakes stepper motor once in specified direction.
+  Returns nothing.
+
+  Param: step_direction
+    Passing step_direction=True opens Airbrakes one step
+    Passing step_direction=False closes Airbrakes one step
+
+    Note: If passing True closes Airbrakes, invert the value of 
+    step_direction when initializing Airbrakes object
+
+calibrate(): 
+  Sets __max_pot_val by reading potentiometer in the fully open state
+  and sets __min_pot_val by reading the potentiometer in the closed state
+  Returns nothing.
+
+  Must be called before launch (on the launch pad) to calibrate the system.
+
+deployBrakes():
+  Opens the brakes to the specifed percentage of fully open.
+
+  Note: This uses the potentiometer for feedback so there will be a small
+  error associated with this. Adjust this error by changing max_error in
+  method body.
+
+  Returns value of potentiometer at final position.
+
+"""
+class Airbrakes:
+  
+  # Set when calibrate() is called
+  __max_pot_val = None
+  __min_pot_val = None
+
+  def __init__(self,  stepper_pin, direction_pin, direction: bool,
+               ads_pot_pin=0, step_angle=1.8, microsteps=16,
+               step_delay_micro_sec=75, gear_box_ratio=14):
+
+    # GPIO pins
+    self.step_pin = stepper_pin
+    self.dir_pin  = direction_pin
+
+    # Stepper motor specs
+    self.motor_direction: bool = direction
+    self.step_delay = step_delay_micro_sec/1000000.0
+    self.step_angle = step_angle
+    self.microsteps = microsteps
+    self.gear_ratio = gear_box_ratio
```

Comment:
I agree with your stance on descriptive variable names but I think this might actually hurt clarity. I do several calculations in the code that would be 150+ characters long with names like that. I do my best to keep lines to 80c max, so I'd have to wrap them which makes things confusing.

And I know you don't agree with the 80c thing but it is actually really annoying when they are longer because it runs off the screen with the way I (and I'm assuming other people) set up their editors. I believe it is even in Googles code standards, but I'll leave the final call up to you.

Reply:
I checked out the google docs and you were right about the 80c. I still think this should be an object really for readability. 

If you do something like `self.motor_options` calling something would only be `self.motor_options['direction']` which isnt 80c. 

Heres one of the cleanest files i've seen at my coop so far 
![image](https://user-images.githubusercontent.com/45109638/158121240-eb85740c-bf1c-4622-bb13-d312db950cd1.png)

See how long those names are lol. As a reviewer I was able to understand the code a lot faster then if I had to read docs or dm the author.

Reply:
This is what I was referring to when I said 150c lines: `    self.__max_steps_to_open = 1.25*(self.stepper_motor_options.gear_ratio*(360/self.stepper_motor_options.step_angle)*self.stepper_motor_options.microsteps)/6`
to me is less readable than the current code: `    self.__max_steps_to_open = 1.25*(self.gear_ratio*(360/self.step_angle)*self.microsteps)/6`

Final call, I'll implement it if you still think its better to have a struct.

Reply:
Sorry if I'm over*stepping* here, but I had a few thoughts on the readability. I realise I'm new to the project and don't have a lot of reputation/authority on code style.

If you want the longer variable names & readability. You could do something like:
```py
# Calculate the motor steps before factoring-in safety and scaling
unscaled_steps = (self.stepper_motor_options.gear_ratio
                  * (360 / self.stepper_motor_options.step_angle)
                  * self.stepper_motor_options.microsteps)

self.__max_steps_to_open = 1.25 * (unscaled_steps / 6)
```

or if you don't want intermediate variables:
```python
# Way one without intermediate variables
self.__max_steps_to_open = 1.25 * (
    self.stepper_motor_options.gear_ratio * (
        360 / self.stepper_motor_options.step_angle
    )
) / 6

# Way two without intermediate variables
self.__max_steps_to_open = (1.25
                            * (self.stepper_motor_options.gear_ratio
                               * (360 / self.stepper_motor_options.step_angle)
                               * self.stepper_motor_options.microsteps)
                            / 6)
```
You can get 80 column and longer variables at the same time in few different ways.

Reply:
Yeah this is a nice middle ground. I'll implement it right now.